core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java

```diff
@@ -0,0 +1,291 @@
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ShortCircuitBlockWriteHandlerTest {
+
+  @Rule
+  public TemporaryFolder mTempFolder = new TemporaryFolder();
+
+  private TestResponseObserver mResponseObserver;
+  private TestBlockWorker mBlockWorker;
+  private ShortCircuitBlockWriteHandler mHandler;
+
+  @Before
+  public void before() throws Exception {
+    File tmpDir = mTempFolder.newFolder();
+    mBlockWorker = new TestBlockWorker(tmpDir.getAbsolutePath());
+
+    mResponseObserver = new TestResponseObserver();
+
+    mHandler = new ShortCircuitBlockWriteHandler(mBlockWorker, mResponseObserver);
+  }
+
+  @Test
+  public void createBlock() {
+    long blockId = 1L;
+    CreateLocalBlockRequest request = createRequest(blockId, false);
+    // request to create a new local block
+    mHandler.onNext(request);
+
+    // verify that we get one response from the handler
+    // that contains the right path to the local block file
+    assertEquals(1, mResponseObserver.getResponses().size());
+    assertNull(mResponseObserver.getError());
+    assertFalse(mResponseObserver.ismCompleted());
+
+    String path = mResponseObserver.getResponses().get(0).getPath();
+    // verify that the local file exists
+    assertTrue(Files.exists(Paths.get(path)));
+
+    // verify that the blockId is recorded into the temp blocks
+    // but is not publicly available yet
+    assertTrue(mBlockWorker.isTempBlockCreated(blockId));
+    assertFalse(mBlockWorker.isBlockCommitted(blockId));
+
+    mHandler.onCompleted();
+
+    // verify that the block is committed and publicly available
+    assertTrue(mBlockWorker.isBlockCommitted(blockId));
+  }
+
+  @Test
+  public void nextRequestWithoutCleaningUp() {
+    long blockId = 1L;
+    CreateLocalBlockRequest request = createRequest(blockId, false);
+    mHandler.onNext(request);
+    long anotherBlockId = 2L;
+    CreateLocalBlockRequest anotherRequest = createRequest(anotherBlockId, false);
+    // this request should fail and abort the previous session
+    mHandler.onNext(anotherRequest);
+
+    assertNotNull(mResponseObserver.getError());
+    assertFalse(mResponseObserver.ismCompleted());
+    assertFalse(mBlockWorker.isTempBlockCreated(blockId));
+    assertFalse(mBlockWorker.isTempBlockCreated(anotherBlockId));
+  }
+
+  @Test
+  public void requestSpaceForNonExistingBlock() {
+    long blockId = 1L;
+    CreateLocalBlockRequest request = createRequest(blockId, true);
+    // this request should fail as the block
+    // is not created yet
+    mHandler.onNext(request);
+
+    assertTrue(mResponseObserver.getResponses().isEmpty());
+    assertFalse(mResponseObserver.ismCompleted());
+    Throwable t = mResponseObserver.getError();
+    assertTrue(
+        t != null
+        && t.getMessage().contains("does not exists"));
+  }
```

Comment:
I think we do not have to inspect the exception. Only assert that an exception is thrown.

Reply:
Fixed